vesta/services/clash_of_code_helper.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import logging
+import uuid
+from typing import Optional
+
+import discord
+import requests
+from discord.utils import MISSING
+from sqlalchemy import select
+
+from . import ClashOfCodeGame, State
+
+BASE_ENDPOINT = "https://www.codingame.com/services/ClashOfCode"
+logger = logging.getLogger(__name__)
+
+def game_id_from_link(link: str) -> str:
+    return link.split("/")[-1]
+
+def fetch(game_id) -> Optional[ClashOfCodeGame]:
+    """
+    Retrieves a game object from the API
+
+    :param game_id: The game id to retrieve
+    :return: The game object if it exists, None otherwise
+    """
+    r = requests.post(f"{BASE_ENDPOINT}/findClashByHandle", json=[
+        game_id
+    ])
+
+    if r.status_code != 200:
+        return None
+
+    return ClashOfCodeGame(**r.json())
+
+def update(game: ClashOfCodeGame) -> ClashOfCodeGame:
+    """
+    Updates the game object with the latest information from the API
+
+    :param game: The game object to update
+    :return: The updated game object for chaining convenience
+    :raises NameError: If there was an error updating the game object
+    """
+    r = requests.post(f"{BASE_ENDPOINT}/findClashByHandle", json=[
+        game.id
+    ])
+
+    if r.status_code != 200:
+        raise NameError("There was an error updating the game object")
+
+    game.hydrate(**r.json())
+    return game
+
+def resume_update_loops() -> None:
+    """
+    Resumes all update loops for all guilds
+
+    :return: Nothing
+    """
+    from .. import session_maker, vesta_client
```

Comment:
Weird imports place

Reply:
If imported at the top of the file, creates a circular import, so it's a workaround that works to avoid it, needs more debugging to find where the circular import takes place